product_sequence/models/product_category.py

```diff
@@ -38,22 +38,53 @@ def _prepare_ir_sequence(self, prefix):
         }
         return vals
 
+    def _get_or_create_sequence(self, prefix):
+        seq_vals = self._prepare_ir_sequence(prefix)
+        sequence = self.env["ir.sequence"].search(
+            [("code", "=", seq_vals.get("code")), ("prefix", "=", prefix)]
+        )
+        if not sequence:
```

Comment:
```suggestion
        sequence = self.env["ir.sequence"].search(
            [("code", "=", seq_vals.get("code")), ("prefix", "=", prefix)]
        )
        if not sequence:
                    seq_vals = self._prepare_ir_sequence(prefix)
```

Reply:
Actually we need `seq_vals.get("code")` to perform the search. So I guess calling the prepare function is enough instead of adding another hook for the "code" preparation or having the string defined twice.